tests/unit/features/RandomFourierDotFeatures_unittest.cc

```diff
@@ -176,3 +176,30 @@ TEST(RandomFourierDotFeatures, add_to_dense_test)
 
 }
 
+TEST(RandomFourierDotFeatures, random_coeff_dim)
+{
+	int32_t num_dims = 50;
+	int32_t vecs = 5;
+	int32_t D = 100;
+
+	SGMatrix<int32_t> data(num_dims, vecs);
+	for (index_t i=0; i<vecs; i++)
+	{
+		for (index_t j=0; j<num_dims; j++)
+			data(j,i) = i+j;
+	}
+
+	auto d_feats = std::make_shared<DenseFeatures<int32_t>>(data);
+	SGVector<float64_t> params(1);
+	params[0] = 8;
+	auto r_feats = std::make_shared<RandomFourierDotFeatures>(
+			d_feats, D, GAUSSIAN, params);
+	
+	auto rows = r_feats->get_random_coefficients().num_rows;
```

Comment:
actually I think this is maybe not the best thing to unit test. You see there can be many ways to compute the embedding, but what counts is that the dimension of the embedding is correct, rather than some internal method

Reply:
Actually, I agree. I've added an ```ASSERT``` in  ```generate_randm_param_vector()```, so it does the job for us.

Reply:
And since it handles the KERNEL based switching(to be modified) it acts like a parameterized test as well. So I guess this test doesn't serve much purpose.

Reply:
I dont think you fixed an error here?

Reply:
ok, again. You need to check the dimension of the **transformed** data. Not the coefficient dimensions. It is better to test interface input/output, not internal states.

Reply:
My bad, I took it for random_coef matrix. About my second comment, I want a unit test which generates random coefficients rather than have it predefined. So, in the added test I created a unit test which uses the said method and checks for the output dimensions, that won't be a problem no?